app/code/Magento/Directory/Block/Data.php

```diff
@@ -172,11 +172,18 @@ public function getRegionCollection()
     /**
      * Returns region html select
      *
+     * @param null|string $defValue
+     * @param string $name
+     * @param string $id
+     * @param string $title
      * @return string
      */
-    public function getRegionHtmlSelect()
+    public function getRegionHtmlSelect($defValue = null,$name = 'region', $id = 'state', $title = 'State/Province')
```

Comment:
Adding new parameters to public method is violate backward compatibility.
I think we can maximize backward compatibility with:
- mark method **getRegionHtmlSelect** as deprecated
- create new method with parameters (e.g. **getRegionSelect** or **getRegionHtml**)
- add `@see` section and call newly created method in **getRegionHtmlSelect**

Reply:
Please rename parameter `$defValue` to `$defaultValue` or just `$value`. 
It is better to avoid reductions.

Reply:
> Adding new parameters to public method is violate backward compatibility.
> I think we can maximize backward compatibility with:
> 
>     * mark method **getRegionHtmlSelect** as deprecated
> 
>     * create new method with parameters (e.g. **getRegionSelect** or **getRegionHtml**)
> 
>     * add `@see` section and call newly created method in **getRegionHtmlSelect**

As per this solution:

How can I pass parameter to **getRegionSelect** if it's parent function (e.g. **getRegionHtmlSelect** ) don't have parameter.

and as all parameters are optional it will work if we don't pass any parameter.

Reply:
The idea that for everyone who used `getRegionHtmlSelect` before, for them nothing will be changed after they will upgrade to a new version of Magento 2 and receive your changes.
Their project will continue working without any troubles and they will be able to refactor their code and use new method `getRegionSelect` which will be introduced by you.

I agree that in general adding optional parameters will work in case of extending of the current class, but this is a violation to [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/).
So, I need to ask you to proceed with the new method and deprecation of the old one.

Reply:
I have done the required changes and pushed them to 2.3-develop branch.